src/LogicTasks/Semantics/Decide.hs

```diff
@@ -61,14 +61,15 @@ instance Show Choice where
 genDecideInst :: DecideConfig -> Gen DecideInst
 genDecideInst DecideConfig{..} = do
     let percentTrueEntries' = fromMaybe (0, 100) percentTrueEntries
-
+    -- jscpd:ignore-start
     formula <- case formulaConfig of
       (FormulaArbitrary syntaxTreeConfig) ->
-        InstArbitrary <$> genSynTree syntaxTreeConfig  `suchThat` withRatio percentTrueEntries'
+        InstArbitrary <$> genSynTree syntaxTreeConfig  `suchThat` \t -> withRatio percentTrueEntries' t && not (hasUnusedAtoms t)
       (FormulaCnf cnfCfg) ->
         InstCnf <$> genCnf' cnfCfg `suchThat` withRatio percentTrueEntries'
       (FormulaDnf dnfCfg) ->
         InstDnf <$> genDnf' dnfCfg `suchThat` withRatio percentTrueEntries'
```

Comment:
What about these other "branches" of the generator? Cannot they also in principle generate formulas that have the same "problem"?